File: Venmo/src/test/java/com/braintreepayments/api/venmo/VenmoAccountUnitTest.java
package com.braintreepayments.api.venmo;

import static junit.framework.Assert.assertEquals;

import com.braintreepayments.api.core.IntegrationType;
import com.braintreepayments.api.core.PaymentMethod;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

@RunWith(RobolectricTestRunner.class)
public class VenmoAccountUnitTest {

    @Test
    public void correctlyBuildsVenmoVaultRequest() throws JSONException {
        VenmoAccount sut = new VenmoAccount("some-nonce", null, PaymentMethod.DEFAULT_SOURCE, IntegrationType.CUSTOM);

        JSONObject fullJson = sut.buildJSON();
        JSONObject venmoAccountJson = fullJson.getJSONObject("venmoAccount");
        assertEquals("some-nonce", venmoAccountJson.getString("nonce"));
    }
}

File: Venmo/src/test/java/com/braintreepayments/api/venmo/VenmoAccountUnitTest.kt
package com.braintreepayments.api.venmo

import com.braintreepayments.api.core.IntegrationType
import com.braintreepayments.api.core.PaymentMethod
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import kotlin.test.Test
import kotlin.test.assertEquals

@RunWith(RobolectricTestRunner::class)
class VenmoAccountUnitTest {

    @Test
    fun `correctly builds Venmo vault Request`() {
        val sut = VenmoAccount("some-nonce", null, PaymentMethod.DEFAULT_SOURCE, IntegrationType.CUSTOM)
        val fullJson = sut.buildJSON()
        val venmoAccountJson = fullJson.getJSONObject("venmoAccount")
        assertEquals("some-nonce", venmoAccountJson.getString("nonce"))
    }
}
